sdks/java/io/iceberg/src/test/java/org/apache/beam/sdk/io/iceberg/IcebergIOLT.java

```diff
@@ -0,0 +1,213 @@
+            options.getTempLocation(), testName.getMethodName(), UUID.randomUUID());
+
+    tableId =
+        TableIdentifier.of(
+            testName.getMethodName(), "table" + Long.toString(UUID.randomUUID().hashCode(), 16));
+  }
+
+  @Test
+  public void testReadWriteHadoopType() {
+    Configuration catalogHadoopConf = new Configuration();
+    catalogHadoopConf.set("fs.gs.project.id", options.getProject());
+    catalogHadoopConf.set("fs.gs.auth.type", "SERVICE_ACCOUNT_JSON_KEYFILE");
+    catalogHadoopConf.set(
+        "fs.gs.auth.service.account.json.keyfile", System.getenv("GOOGLE_APPLICATION_CREDENTIALS"));
+
+    Catalog catalog = new HadoopCatalog(catalogHadoopConf, warehouseLocation);
+    catalog.createTable(tableId, icebergSchema);
+
+    Map<String, Object> config =
+        ImmutableMap.<String, Object>builder()
+            .put("table", tableId.toString())
+            .put(
+                "catalog_config",
+                ImmutableMap.<String, String>builder()
+                    .put("catalog_name", "hadoop")
+                    .put("catalog_type", CatalogUtil.ICEBERG_CATALOG_TYPE_HADOOP)
+                    .put("warehouse_location", warehouseLocation)
+                    .build())
+            .build();
+
+    PCollection<Long> source =
+        writePipeline.apply(GenerateSequence.from(0).to(testConfiguration.numRows()));
+
+    PCollection<Row> inputRows =
+        source
+            .apply(
+                "Create Rows",
+                MapElements.into(TypeDescriptors.rows())
+                    .via(
+                        new GenerateRow(
+                            testConfiguration.numFields(),
+                            testConfiguration.byteSizePerField(),
+                            beamSchema)))
+            .setRowSchema(beamSchema);
+
+    PCollectionRowTuple.of("input", inputRows)
+        .apply(Managed.write(Managed.ICEBERG).withConfig(config));
+    writePipeline.run().waitUntilFinish();
+
+    // read pipeline
+    PCollection<Long> countRows =
+        PCollectionRowTuple.empty(readPipeline)
+            .apply(Managed.read(Managed.ICEBERG).withConfig(config))
+            .get("output")
+            .apply(Count.globally());
+
+    PAssert.thatSingleton(countRows).isEqualTo(testConfiguration.numRows());
+    readPipeline.run().waitUntilFinish();
+  }
+
+  private static class GenerateRow implements SerializableFunction<Long, Row> {
+    final int numFields;
+    final int byteSizePerField;
+    final org.apache.beam.sdk.schemas.Schema schema;
+
+    GenerateRow(int numFields, int byteSizePerField, org.apache.beam.sdk.schemas.Schema schema) {
+      this.numFields = numFields;
+      this.byteSizePerField = byteSizePerField;
+      this.schema = schema;
+    }
+
+    @Override
+    public Row apply(Long input) {
+      Row.Builder rowBuilder = Row.withSchema(schema);
+      for (int i = 0; i < numFields; i++) {
+        rowBuilder = rowBuilder.addValue(new byte[byteSizePerField]);
```

Comment:
Would it be meaningful to populate the field with random data? I don't know if there might be some points in the execution where because it is all zeros it gets super compressed so the results are not representative.

Reply:
Yeah good point. Will fill it up with random bytes instead